### PR TITLE
Update AVIF build to produce shipped wasm

### LINKED ISSUE #805: Smaller AVIF wasms
Right now, both the encoder and the decoder contain both the encoder and decoder.

If I build the encoder with `CONFIG_AV1_DECODER` and `CONFIG_AV1_HIGHBITDEPTH=0` in the AOM build, it knocks around 200k off the wasm.

If I build the decoder with `CONFIG_AV1_ENCODER=0`, it knocks almost 1mb off the wasm.

It complains about missing symbols, so I included `-s ERROR_ON_UNDEFINED_SYMBOLS=0` in the emscripten part, but it seems to function correctly.

I'm not familiar enough with MAKE etc to automate this. @surma @RReverser can you take a look? Best to do it on the `avif-options` branch, as I've made some changes to the cpp there.

### PR DESCRIPTION
**Pull request by jakearchibald**
Fixes #805.

### CODE REVIEW

**Comment by jakearchibald** [comment]
Wait, no this doesn't work. I fucked something up when moving stuff into variables.

**Comment by jakearchibald** [comment]
And now it's working.

File: codecs/avif/Makefile [hunk 1]
```diff
@@ -1,76 +1,139 @@
...
-		-s MODULARIZE=1 \
+		${EMSCRIPTEN_FLAGS} \
 		-s 'EXPORT_NAME="$(basename $(@F))"' \
 		-o $@ \
 		$+
 
-$(CODEC_OUT): $(CODEC_DIR)/CMakeLists.txt $(LIBAOM_OUT)
-	mkdir -p $(CODEC_BUILD_DIR)
-	cd $(CODEC_BUILD_DIR) && \
+# ERROR_ON_UNDEFINED_SYMBOLS=0 is needed to seperate the encoder and decoder
```
Comment (RReverser) [inline codecs/avif/Makefile]: This comment should be moved to the var definition probably.

File: codecs/avif/Makefile [hunk 2]
```diff
@@ -1,76 +1,139 @@
...
+OUT_ENC_CPP = $(OUT_ENC_JS:.js=.cpp)
+OUT_ENC_WASM = $(OUT_ENC_JS:.js=.wasm)
+
+OUT_DEC_JS = dec/avif_dec.js
+OUT_DEC_CPP = $(OUT_DEC_JS:.js=.cpp)
+OUT_DEC_WASM = $(OUT_DEC_JS:.js=.wasm)
 
-OUT_JS = enc/avif_enc.js dec/avif_dec.js
-OUT_WASM = $(OUT_JS:.js=.wasm)
+EMSCRIPTEN_FLAGS = --bind \
```
Comment (RReverser) [inline codecs/avif/Makefile]: Up to you, but I'd probably include `${CXXFLAGS}` and `${LDFLAGS}` in here.

**Comment by jakearchibald** [comment]
@RReverser Fixed those issues. Thanks for the review!

**Comment by surma** [comment]
If this LGT Ingvar, then LGTM :D

**Review by RReverser — APPROVED** [review]

**Pull request merged**